spec/requests/health_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Health Controller', type: :request do
+  describe 'GET #show' do
+    context 'when the application is up' do
+      it 'returns a successful response with HTML' do
+        get '/'
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('text/html; charset=utf-8')
+        expect(response.body).to include('<html><body style="background-color: green"></body></html>')
+      end
+
+      it 'returns a successful response with JSON' do
+        get '/', headers: { 'ACCEPT' => 'application/json' }
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response.body).to eq({ message: 'OK' }.to_json)
+      end
+    end
+
+    context 'when the application is down' do
+      before(:each) do
+        error = StandardError.new('Something went wrong')
+        allow_any_instance_of(HealthController).to receive(:show).and_raise(error)
```

Comment:
`allow_any_instance_of` is a dirty way to achieve this 🙈 See a better mocking approach: https://stackoverflow.com/a/27144891